framework/platform/wayland/tcuWayland.cpp

```diff
@@ -0,0 +1,223 @@
+/*-------------------------------------------------------------------------
+ * drawElements Quality Program Tester Core
+ * ----------------------------------------
+ *
+ * Copyright 2014 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *//*!
+ * \file
+ * \brief wayland utilities.
+ *//*--------------------------------------------------------------------*/
+
+#include "tcuWayland.hpp"
+#include "gluRenderConfig.hpp"
+#include "deMemory.h"
+
+#include <stdio.h>
+
+namespace tcu
+{
+namespace wayland
+{
+
+enum
+{
+	DEFAULT_WINDOW_WIDTH	= 400,
+	DEFAULT_WINDOW_HEIGHT	= 300
+};
+
+EventState::EventState (void)
+	: m_quit(false)
+{
+}
+
+EventState::~EventState (void)
+{
+}
+
+void EventState::setQuitFlag (bool quit)
+{
+	de::ScopedLock lock(m_mutex);
+	m_quit = quit;
+}
+
+bool EventState::getQuitFlag (void)
+{
+	de::ScopedLock lock(m_mutex);
+	return m_quit;
+}
+const struct wl_registry_listener Display::s_registryListener =
+{
+	Display::handleGlobal,
+	Display::handleGlobalRemove
+};
+
+const struct wl_shell_surface_listener Window::s_shellSurfaceListener =
+{
+	Window::handlePing,
+	Window::handleConfigure,
+	Window::handlePopupDone,
+};
+
+void Display::handleGlobal(void *data,struct wl_registry *registry, uint32_t id, const char *interface, uint32_t version UNUSED)
```

Comment:
This code appears to violate coding guidelines in a few places (placement of *), whitespace, ..